yodalib/decompilers/ghidra/interface.py

```diff
@@ -270,6 +270,39 @@ def _structs(self) -> Dict[str, Struct]:
             name: Struct(name, size, members=self._struct_members_from_gstruct(name)) for name, size in name_sizes
         } if name_sizes else {}
 
+    @ghidra_transaction
+    def _set_enum(self, enum: Enum, **kwargs) -> bool:
+        corrected_enum_name = "/" + enum.name
+        old_ghidra_enum = self.ghidra.currentProgram.getDataTypeManager().getDataType(corrected_enum_name)
+        data_manager = self.ghidra.currentProgram.getDataTypeManager()
+        handler = self.ghidra.import_module_object("ghidra.program.model.data", "DataTypeConflictHandler")
+        enumType = self.ghidra.import_module_object("ghidra.program.model.data", "EnumDataType")
+        categoryPath = self.ghidra.import_module_object("ghidra.program.model.data", "CategoryPath")
+        ghidra_enum = enumType(categoryPath('/'), enum.name, 4)
+        for m_name, m_val in enum.members.items():
+            ghidra_enum.add(m_name, m_val)
+
+        try:
+            if old_ghidra_enum:
+                data_manager.replaceDataType(old_ghidra_enum, ghidra_enum, True)
+            else:
+                data_manager.addDataType(ghidra_enum, handler.DEFAULT_HANDLER)
+            return True
+        except Exception as ex:
+            print(f'Error adding enum {enum.name}: {ex}')
+            return False
+
+    def _get_enum(self, name) -> Optional[Enum]:
+        return Enum(name, self._get_enum_members('/' + name))
+
+    def _enums(self) -> Dict[str, Enum]:
+        names = [
```

Comment:
Do you happen to know if this is fast or slow? I.e., should it be remote, or is it fast enough?

Reply:
Its been ok because enum count has been limited, but its pretty slow per enum

Reply:
Not sure how to do multi-line remote stuff though

Reply:
@Flipout50 but this list comprehension is not multi line. It can be in one line

Reply:
Oh yeah ok, its probably best to make remote then, ill make sure to fix that